### PR TITLE
[173] Change pin/unpin button activation policy with multiple selection

### PR DESCRIPTION
**Pull request by scosta-obeo**
The pin/unpin button is now checked only if all selected pinnable elements are pinned.

Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/205

### CODE REVIEW

**Comment by scosta-obeo** [comment]
This commit also changes the listeners from async to sync to prevent the action from being disposed of due to an asynchronous call.

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java [hunk 1]
```diff
@@ -187,48 +170,57 @@ private void updateActionState(ISelection selection) {
...
-            pinListener = null;
-
-            Resource resource = element.eResource();
-            if (resource != null) {
-                var permissionAuthoriry = PermissionAuthorityRegistry.getDefault().getPermissionAuthority(resource.getResourceSet());
-                permissionAuthoriry.removeAuthorityListener(authorityListener);
+        if (selectionElements != null) {
+            for (Entry<DDiagramElement, ResourceSetListener> entry : selectionElements.entrySet()) {
+                DDiagramElement element = entry.getKey();
+                ResourceSetListener pinListeners = entry.getValue();
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java:203]: pinListener without s

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java [hunk 2]
```diff
@@ -187,48 +170,57 @@ private void updateActionState(ISelection selection) {
...
+                    ResourceSetListener pinListener = new PinHelper().createPinListener(elem, () -> {
+                        EclipseUIUtil.displayAsyncExec(this::updatePinState);
+                    });
+                    domain.addResourceSetListener(pinListener);
+
+                    Resource resource = elem.eResource();
+                    if (resource != null) {
+                        var permissionAuthoriry = PermissionAuthorityRegistry.getDefault().getPermissionAuthority(resource.getResourceSet());
+                        permissionAuthoriry.addAuthorityListener(authorityListener);
+                    }
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java:186]: it seems that you do not need to add and remove as many authority listener as selected elements

**Review by lfasani — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java (hunk 2 above)
Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/pinning/PinElementsAction.java]: Ok, I now have one listener per ResourceSet (only one in most cases)

**Review by lfasani — APPROVED** [review]

**Pull request merged**